fullcalendar-app/src/components/FullCalendar.vue

```diff
@@ -215,23 +250,34 @@ export default {
         clone.querySelectorAll('.fc-listDay-view tbody td')
           .forEach(function(element) {
             element.style.height = '3rem';
-        });
+          });
       }
-
-      clone.querySelectorAll('.fullcalendar--header, .fc-header-toolbar .fc-toolbar-chunk:not(:first-child), .calendar-branch-info h4')
-        .forEach(function(element) {
+    },
+    cleanupUnwantedElements(clone) {
+      const elementsToHide = clone.querySelectorAll(
+        '.fullcalendar--header, .fc-header-toolbar .fc-toolbar-chunk:not(:first-child), .calendar-branch-info h4'
+      );
+      elementsToHide.forEach(element => {
         element.style.display = 'none';
       });
-
-      // Adding copy to the DOM for PDF generation. The copy will be deleted immediately, so the user will not see the change.
+    },
+    addCloneToDOM(clone) {
       document.body.appendChild(clone);
+    },
+    generatePDF(clone) {
+      const scheduleTitle = document.querySelector('.fc-toolbar-title').textContent;
+      const filename = `Weekly schedule (${scheduleTitle}).pdf`;
 
-      // Setting parameters for html2pdf.
-      const options = {
+      const pdfOptions = {
         margin: [3, 5],
-        filename: fileName,
-        image: { type: 'jpeg', quality: 0.99 },
-        html2canvas: { scale: 2, useCORS: true },
+        filename: filename,
+        image: { type: 'jpeg', quality: 0.98 },
```

Comment:
So we are packing image, not vector format of HTML?
Can we convert HTML to PDF?

Reply:
I think that this is how the html2pdf library works, i.e. first the canvas is created and then the PDF. This allows us to take a "snapshot" of the DOM element with everything we need, i.e. styles and javascript, otherwise manipulation of the PDF document will probably be even more difficult.